skimage/segmentation/_join.py

```diff
@@ -121,24 +121,21 @@ def relabel_sequential(label_field, offset=1):
         raise ValueError("Offset must be strictly positive.")
     if np.min(label_field) < 0:
         raise ValueError("Cannot relabel array that contains negative values.")
-    m = label_field.max()
+    max_label = label_field.max()
     if not np.issubdtype(label_field.dtype, np.integer):
-        new_type = np.min_scalar_type(int(m))
+        new_type = np.min_scalar_type(int(max_label))
         label_field = label_field.astype(new_type)
-        m = m.astype(new_type)  # Ensures m is an integer
+        max_label = max_label.astype(new_type)  # Ensures max_label is an integer
     labels = np.unique(label_field)
     labels0 = labels[labels != 0]
-    required_type = np.min_scalar_type(offset + len(labels0))
+    new_max_label = offset - 1 + len(labels0)
+    new_labels0 = np.arange(offset, int(new_max_label + 1))
+    required_type = np.min_scalar_type(new_max_label)
     if np.dtype(required_type).itemsize > np.dtype(label_field.dtype).itemsize:
         label_field = label_field.astype(required_type)
-    new_labels0 = np.arange(offset, offset + len(labels0))
-    if np.all(labels0 == new_labels0):
-        return label_field, labels, labels
-    forward_map = np.zeros(int(m + 1), dtype=label_field.dtype)
+    forward_map = np.zeros(int(max_label + 1), dtype=label_field.dtype)
```

Comment:
I think that you can directly use `new_type` here and avoid the `if` above
```suggestion
    forward_map = np.zeros(int(max_label + 1), dtype=new_type)
```

Reply:
> I think that you can directly use `required_type` here and avoid the `if` above

In my last pull request (#3740), I changed the function to retain the data type of the input `label_field`. Hence, I would keep that. But you are right that it is not necessary to convert (i.e. make a copy of) `label_field` to the new data type. I changed that in the latest commit.